### ISSUE TITLE
Resource Optimization: Performance - Division By Zero Error

### ISSUE THREAD

**Issue by sfc-gh-twhite** [Snowflake-Labs/sfquickstarts#449]
**Describe the bug**
In the **4. Scale Up vs. Out (Size vs. Multi-cluster) (T2).** section in [Resource Optimization: Performance](https://quickstarts.snowflake.com/guide/resource_optimization_performance_optimization/index.html) there is potential for the 
`--LIST OF WAREHOUSES AND QUERIES WHERE A LARGER WAREHOUSE WOULD HAVE HELPED WITH REMOTE SPILLING` query to fail with a `Division by zero` error. 

**URL of where you see the bug**
https://quickstarts.snowflake.com/guide/resource_optimization_performance_optimization/index.html

**To Reproduce**
Run the following query in an account that has at least one query that has scanned zero bytes. 

```
SELECT QUERY_ID
,USER_NAME
,WAREHOUSE_NAME
,WAREHOUSE_SIZE
,BYTES_SCANNED
,BYTES_SPILLED_TO_REMOTE_STORAGE
,BYTES_SPILLED_TO_REMOTE_STORAGE / BYTES_SCANNED AS SPILLING_READ_RATIO
FROM "SNOWFLAKE"."ACCOUNT_USAGE"."QUERY_HISTORY"
WHERE BYTES_SPILLED_TO_REMOTE_STORAGE > BYTES_SCANNED * 5  -- Each byte read was spilled 5x on average
ORDER BY SPILLING_READ_RATIO DESC
;
```

**Additional context**
Would look to resolve this by adding an additional predicate in the WHERE clause to look at `BYTES_SCANNED > 0` which should still provide correct and beneficial results.

**Comment by sfc-gh-kszajowski** [comment]
Instead of using a where clause, can we utilize DIV0 in the select statement?

**Comment by sfc-gh-twhite** [comment]
[DIV0](https://docs.snowflake.com/en/sql-reference/functions/div0.html) would suffice and avoid adding the additional predicate. If someone sees that the `SPILLING_READ_RATIO` = 0, it would likely catch someone's attention.

**Comment by sfc-gh-clathrope** [comment]
I agree with idea of adding BYTES_SCANNED > 0 and a START_TIME filter of last 2 weeks or similar. This will eliminate many of the queries that do metadata operations or write only data loads. Without it, customers with decent amount of metadata operations will have an artificially lowered remote scan byte ratio due to these rows that really is not relevant to the purpose of warehouse size (they don't even use a warehouse in the non-write scenario). Also, 365 days of query_history without a start_time filter is way too long a period to analyze as workloads shift over time and will need revisiting on a periodic basis.
